homeassistant/components/nexia/climate.py

```diff
@@ -224,20 +234,43 @@ def preset_mode(self):
         return self._zone.get_preset()
 
     async def async_set_humidity(self, humidity: int) -> None:
-        """Dehumidify target."""
-        if self._thermostat.has_dehumidify_support():
-            await self.async_set_dehumidify_setpoint(humidity)
+        """Set Dehumidify target."""
+
+        if self._zone.get_current_mode() == OPERATION_MODE_HEAT:
+            if self._thermostat.has_humidify_support():
+                await self.async_set_humidify_setpoint(humidity)
+        elif self._zone.get_current_mode() == OPERATION_MODE_COOL:
+            if self._thermostat.has_dehumidify_support():
+                await self.async_set_dehumidify_setpoint(humidity)
         else:
-            await self.async_set_humidify_setpoint(humidity)
+            if self._thermostat.has_humidify_support():
+                await self.async_set_humidify_setpoint(humidity)
+            if self._thermostat.has_dehumidify_support():
+                await self.async_set_humidify_setpoint(humidity)
         self._signal_thermostat_update()
 
     @property
-    def target_humidity(self):
-        """Humidity indoors setpoint."""
-        if self._has_dehumidify_support:
-            return percent_conv(self._thermostat.get_dehumidify_setpoint())
-        if self._has_humidify_support:
-            return percent_conv(self._thermostat.get_humidify_setpoint())
+    def target_humidity(self) -> float | None:
+        """Humidity indoors setpoint.
+
+        In systems that support both humidification and dehumidification,
+        two values for target exist. We must choose one to return.
+
+        :return: The target humidity setpoint.
+        """
+        if self._zone.get_current_mode() == OPERATION_MODE_HEAT:
+            if self._has_humidify_support:
+                return percent_conv(self._thermostat.get_humidify_setpoint())
+        if self._zone.get_current_mode() == OPERATION_MODE_COOL:
+            if self._has_dehumidify_support:
+                return percent_conv(self._thermostat.get_dehumidify_setpoint())
+        else:
+            # Fall back to previous behavior of returning dehumidify value then humidify
+            if self._has_dehumidify_support:
+                return percent_conv(self._thermostat.get_dehumidify_setpoint())
+            if self._has_humidify_support:
+                return percent_conv(self._thermostat.get_humidify_setpoint())
+
```

Comment:
```suggestion
        # If heat is on, always return humidify value first
        if self._zone.get_current_mode() == OPERATION_MODE_HEAT and self._has_humidify_support:
            return percent_conv(self._thermostat.get_humidify_setpoint())
        # Fall back to previous behavior of returning dehumidify value then humidify
        if self._has_dehumidify_support:
            return percent_conv(self._thermostat.get_dehumidify_setpoint())
        if self._has_humidify_support:
            return percent_conv(self._thermostat.get_humidify_setpoint())
```

Maybe something like this